src/controller/Controller.js

```diff
@@ -0,0 +1,62 @@
+const BridgeInteractor = require('../domain/usecases/BridgeInteractor');
+const UserInteractor = require('../domain/usecases/UserInteractor');
+const ViewInteractor = require('../domain/usecases/ViewInteractor');
+
+const BridgeGame = require('../service/BridgeGame');
+const STATUS = require('../service/service.constants');
+
+class Controller {
+  constructor() {
+    const user = new UserInteractor();
+    const bridge = new BridgeInteractor();
+    this.service = new BridgeGame(user, bridge);
+    this.viewer = new ViewInteractor(user, bridge);
+    this.BehaviorHash = this.setStateBehavior();
+  }
+
+  setStateBehavior() {
+    return new Map([
+      [STATUS.done, this.exit.bind(this, true)],
+      [STATUS.running, this.playRoutine.bind(this)],
+      [STATUS.failure, this.retry.bind(this, false)],
+    ]);
+  }
+
+  play() {
+    this.viewer.getBridgeSize((size) => {
+      this.service.makeBridge(size);
+      this.playRoutine();
+    });
+  }
+
+  playRoutine() {
+    this.viewer.getMove((direction) => {
+      this.move(direction);
+      const { status } = this.service.getGameStatus();
+      this.BehaviorHash.get(status)();
+    });
+  }
+
+  move(direction) {
+    this.service.move(direction);
+    this.viewer.printMap();
+  }
+
+  retry() {
+    this.viewer.getGameCommand((command) => {
+      this.service.retry(command, {
+        restart: () => {
+          this.service.resetUser();
+          this.playRoutine();
+        }, exit: this.exit.bind(this),
+      });
+    });
+  }
```

Comment:
이 부분은 인자값으로 Object 형태가 들어가고, 안에 restart, exit 함수가 들어갔네요
restart도 exit처럼 메서드로 분리해서 적는게 어떨까요?
```suggestion
  retry() {
    this.viewer.getGameCommand((command) => {
      this.service.retry(command, {
        restart: this.restart.bind(this), exit: this.exit.bind(this),
      });
    });
  }
```

Reply:
아 그렇네요! BridgeGame이 restart변수를 갖도로 만들어줘야 좀더 단순화되겠네요!:) 감사합니다!:)